core/app/models/workarea/data_file/csv.rb

```diff
@@ -48,12 +48,22 @@ def export!
 
       def assign_attributes(model, attrs)
         model.fields.each do |name, metadata|
+          next if name == 'updated_at'
+
           value = CsvFields.deserialize_from(attrs, field: metadata, model: model)
-          model.send("#{name}=", value) if value.present?
+          changed = if value.is_a?(Time)
+                      value.to_i != model[name].to_i
+                    else
+                      value != model[name]
+                    end
+
+          model.send("#{name}=", value) if value.present? && changed
         end
 
-        model.updated_at ||= Time.current
-        model.created_at ||= Time.current
+        if model.embedded? && model.changed?
+          model.updated_at = Time.current
+          model.created_at ||= Time.current
```

Comment:
Putting this in an `if` is a change, we'll always want to ensure it.

Reply:
I was seeing `updated_at` getting changed on records that didn't actually get updated, so for example if you exported a CSV, changed one line, and re-imported it, _all_ of the entries would get updated, which would break a whole lot of cache entries unnecessarily.

Reply:
I mean specifically the `created_at` should not be an `if`, we always want to ensure that's present.

Reply:
wouldn't mongoid do that for us?

Reply:
If you pass an invalid `created_at`, Mongoid will leave it as `nil` iirc :(